### ISSUE TITLE
Error [ERR_REQUIRE_ESM] after use esbuild for cjs

### ISSUE THREAD

**Issue by mengshouer** [Sec-ant/zxing-wasm#49]
![image](https://github.com/Sec-ant/zxing-wasm/assets/36481080/04fe6738-ac13-4f3a-991d-6b8c3b030374)

![image](https://github.com/Sec-ant/zxing-wasm/assets/36481080/52b6b2df-dac0-48ca-b35e-eddc2c42ef32)

```js
// build.js
const { build } = require("esbuild");
const { dependencies } = require("./package.json");

const sharedConfig = {
  entryPoints: ["index.ts"],
  bundle: true,
  minify: true,
  external: Object.keys(dependencies),
};

build({
  ...sharedConfig,
  platform: "node", // for CJS
  outfile: "dist/index.js",
});

```

```
// tsconfig.json
{
  "compilerOptions": {
    "target": "ES2022",
    "module": "commonjs",
    "skipLibCheck": true,
    "moduleResolution": "node"
  },
  "include": ["."]
}
```

I saw the Usage:
> If you use TypeScript, you should set [moduleResolution](https://www.typescriptlang.org/docs/handbook/modules/theory.html#module-resolution) to [bundler](https://www.typescriptlang.org/docs/handbook/modules/reference.html#bundler), [node16 or nodenext](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext-1) in your tsconfig.json file to properly resolve the exported module.

consider compatibility with cjs?

**Comment by Sec-ant** [comment]
I can successfully run the demo in node (with a CommonJs context) with the following configs:

- `package.json`
  ```json
  {
    "name": "zxing-wasm-test",
    "version": "1.0.0",
    "description": "",
    "main": "index.js",
    "scripts": {
      "start": "node -r esbuild-register index.ts"
    },
    "author": "",
    "dependencies": {
      "zxing-wasm": "^1.0.0-rc.7"
    },
    "devDependencies": {
      "esbuild": "^0.19.7",
      "esbuild-register": "^3.5.0",
      "typescript": "^5.3.2"
    }
  }
  ```
- `tsconfig.json`
  ```json
  {
    "compilerOptions": {
      "target": "ES2022",
      "module": "nodenext",
      "skipLibCheck": true,
      "moduleResolution": "nodenext"
    }
  }
  ```

- `index.ts`
  ```ts
  async function run() {
    const { readBarcodesFromImageFile } = await import("zxing-wasm");
  
    const imageFile = await fetch(
      "https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=Hello%20world!"
    ).then((resp) => resp.blob());
  
    const imageFileReadResults = await readBarcodesFromImageFile(imageFile, {
      formats: ["QRCode"],
      maxNumberOfSymbols: 1,
    });
  
    console.log(imageFileReadResults[0].text);
  }
  
  run();
  ```

- `shell`
  ```shell
  > npm run start
  ```

According to the [typescript doc on `moduleResolution`](https://www.typescriptlang.org/tsconfig#moduleResolution)

> 'node10' (previously called 'node') for Node.js versions older than v10, which only support CommonJS require. You probably won’t need to use node10 in modern code.

Is there any reason that you're still using the legacy `node` option instead of `nodenext` here?

> consider compatibility with cjs?

The above example is the way to import this package in a commonjs module context. I admit It's not supported as a first class citizen because dual package publishing can be cumbersome. If you can share your use case maybe I can further think it over.

**Comment by Sec-ant** [comment]
This `tsconfig` also works
```json
{
  "compilerOptions": {
    "target": "ES2022",
    "module": "commonjs",
    "skipLibCheck": true,
    "moduleResolution": "node10"
  }
}
```

**Comment by mengshouer** [comment]
The dev environment is used normally, but errors occur after build with esbuild

**Comment by Sec-ant** [comment]
I made a PR to support `cjs`, can you checkout #50 locally and test if it works?

**Comment by mengshouer** [comment]
I downloaded your file package and it seems to be working

**Comment by Sec-ant** [comment]
Shipped in [v1.0.0-rc.9](https://www.npmjs.com/package/zxing-wasm/v/1.0.0-rc.9) (I made a mistake so don't use rc.8), you can install from npm registry and check again.

**Comment by mengshouer** [comment]
in my project, it works
